doc/api/package.json.md

```diff
@@ -0,0 +1,307 @@
+# Package.json
+
+<!--introduced_in=REPLACEME-->
+<!-- type=misc -->
+
+## Introduction
+
+A folder containing a `package.json` file, and all subfolders below that folder
+down until the next folder containing another `package.json`, is considered a
+_package scope_. This document aims to describe the fields used by Node.js
+runtime.
```

Comment:
```suggestion
_package scope_. This document aims to describe the fields used by the
Node.js runtime.
```

Reply:
The two sentences in this paragraph don't really belong together. The first defines a package scope, the second introduces this document.

The overall order of sections is a bit off, IMO. I would think that `"name"` should come first, and there should be an explanation of how this field is used with `require` and `import`. When one does `require('foo')`, does Node look for a folder named `node_modules/foo` or a package with `"name": "foo"`? What happens when they disagree? Etc. If Node doesn't use the field at all, we shouldn't be documenting the field.

After that I think should be introduction of package scope and `"type"`. Then exports, conditional exports and finally `"main"`.

Reply:
Perhaps in the introduction you can list all the fields that Node recognizes, in alphabetical order, with each field linked to its section? That way we can keep a section name like Conditional Exports, but in the introduction you can have something like “`"exports"`: See [Exports](#) and [Conditional Exports](#)”. And then the sections can be in logical order of what makes the most sense for someone reading this document from top to bottom, but there’s still this table of contents to jump to a particular field.

As for the resolution algorithm, I think it differs in CommonJS versus ESM, so it's probably best kept in each of those sections and linked to from here.

Reply:
> Perhaps in the introduction you can list all the fields that Node recognizes, in alphabetical order, with each field linked to its section?

Wouldn't that be redundant with the TOC?

Reply:
Possibly, though there’s a disparity in the section names like Conditional Exports and the fields like `"exports"`. We probably want to provide a way for people to look them up via either.

Reply:
@GeoffreyBooth I've made some changes, now the TOC look like this:

```markdown
* `package.json`
    * Introduction
    * Supported fields
        * `"name"`
        * `"exports"`
            * Subpath Exports
            * Package Exports Fallbacks
            * Exports Sugar
            * Conditional Exports
            * Nested conditions
        * `"main"`
        * `"type"`
```

Does that work for you?

Reply:
Sure, that looks great!

Reply:
Maybe I'm missing something, but I don't see how `"type"` and `"exports"` are related... `"type"` is atm related to the `"main"` field, but #33086 should change that.

The way I see it, its sole purpose is to set the correct parser for `.js` files, which is different from other fields that affect the resolution algorithm, which is why I put it last.